protocols/identify/src/handler.rs

```diff
@@ -371,8 +468,11 @@ impl ConnectionHandler for Handler {
             ConnectionEvent::FullyNegotiatedOutbound(fully_negotiated_outbound) => {
                 self.on_fully_negotiated_outbound(fully_negotiated_outbound)
             }
-            ConnectionEvent::DialUpgradeError(dial_upgrade_error) => {
-                self.on_dial_upgrade_error(dial_upgrade_error)
+            ConnectionEvent::DialUpgradeError(_dial_upgrade_error) => {
+                self.events.push(ConnectionHandlerEvent::NotifyBehaviour(
+                    Event::IdentificationError(StreamUpgradeError::NegotiationFailed),
```

Comment:
Why hard-code this to `NegotiationFailed` instead of using the `_dial_upgrade_error`?

Reply:
Due to the error
Type mismatch [E0308] expected `StreamUpgradeError<UpgradeError>`, but found `StreamUpgradeError<Either<Void, Void>>`